c10/core/TensorImpl.h

```diff
@@ -1353,9 +1353,31 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
    * WARNING: This is NOT computed in bytes.
    */
   TENSORIMPL_MAYBE_VIRTUAL int64_t storage_offset() const {
+    hould
+    TORCH_CHECK(!has_symbolic_sizes_strides());
     return storage_offset_;
   }
```

Comment:
This needs the policy treatment

Reply:
I was being consciously lazy here. Except for being consistent with the other `TensorImpl` accessors, we don't seem to be getting a lot of benefit from customizing the int version of `storage_offset` since no backend seems to need it (the int version). If we need to return a symint we can't use the int `storage_offset_custom` anyways and hopefully it would go away after we switch to using `sym_storage_offset` exclusively.